tests/unit/spec/base/stub-attribution.js

```diff
@@ -214,6 +214,84 @@ describe('stub-attribution.js', function() {
 
             expect(Mozilla.StubAttribution.hasValidData(data)).toBeFalsy();
         });
+
+        it('should validate double-encoded attribution data correctly', function() {
+            /* eslint-disable camelcase */
+            const data = {
+                utm_source: 'addons.mozilla.org',
+                utm_medium: 'referral',
+                utm_campaign: 'amo-fx-cta-607454',
+                utm_content: 'rta%25253AdUJsb2NrMEByYXltb25kaGlsbC5uZXQ',
+                referrer: 'https://addons.mozilla.org/',
+                ua: 'chrome',
+                visit_id: GA_VISIT_ID
+            };
+            /* eslint-enable camelcase */
+
+            expect(Mozilla.StubAttribution.hasValidData(data)).toBeTruthy();
+
+            /* eslint-disable camelcase */
+            const data2 = {
+                utm_source: 'addons.mozilla.org',
+                utm_medium: 'referral',
+                utm_campaign: 'amo-fx-cta-607454',
+                utm_content: 'rta%25253AdUJsb2NrMEByYXltb25kaGlsbC5uZXQ',
+                referrer: '',
+                ua: 'chrome',
+                visit_id: GA_VISIT_ID
+            };
+            /* eslint-enable camelcase */
+
+            expect(Mozilla.StubAttribution.hasValidData(data2)).toBeFalsy();
+        });
+
+        it('should validate non-encoded attribution data as expected', function() {
+            /* eslint-disable camelcase */
+            const data = {
+                utm_source: 'addons.mozilla.org',
+                utm_medium: 'referral',
+                utm_campaign: 'amo-fx-cta-607454',
+                utm_content: 'rta:dUJsb2NrMEByYXltb25kaGlsbC5uZXQ',
+                referrer: 'https://addons.mozilla.org/',
+                ua: 'chrome',
+                visit_id: GA_VISIT_ID
+            };
+            /* eslint-enable camelcase */
+
+            expect(Mozilla.StubAttribution.hasValidData(data)).toBeTruthy();
+        });
+
+        it('should limit the length of the UTM content for RTAMO data', function() {
+            /* eslint-disable camelcase */
+            const data = {
+                utm_source: 'addons.mozilla.org',
+                utm_medium: 'referral',
+                utm_campaign: 'amo-fx-cta-607454',
+                utm_content: `rta%${'25'.repeat(58)}3AdUJsb2NrMEByYXltb25kaGlsbC5uZXQ`,
+                referrer: 'https://addons.mozilla.org/',
```

Comment:
This could be set to an empty string, right?

Reply:
Why? Would the test then be passing as falsy because of the length, or because of the referrer? Harder to tell imho, unless I'm not understanding why you suggest this?

Reply:
Without the length check, the function would return `true` with _and_ without the `referer` set, which is a problem. Because the added length check is before, it does not matter right now but if there is some refactoring in the future, we might miss this edge case.

Reply:
Changing this value won't make reasonsing about future changes any clearer though, unless I'm not following your trail of thought here? 

If we don't have the length check it will pass validation yes, but we have to set a sensible limit on how many times we're going to try decoding these values without it. I'm not sure why a future refactor would try and move the length check later on in the code, seeing as it would be redundant if not used as an early return. Are we perhaps leaning a little too far into the edgecases here? :)

Reply:
I guess I could add another referrer check along with the length check, so we only return early for non-AMO referrals? Thoughts?

Reply:
Maybe you could add both test cases again, with and without the `referer` set?